src/android/CodePush.java

```diff
@@ -270,6 +270,44 @@ private void cleanOldPackageSilently() {
         }
     }
 
+    private void clearDeploymentsIfBinaryUpdated() {
+        /* check if we have a deployed package already */
+        CodePushPackageMetadata deployedPackageMetadata = this.codePushPackageManager.getCurrentPackageMetadata();
+        if (deployedPackageMetadata != null) {
+            String deployedPackageTimeStamp = deployedPackageMetadata.nativeBuildTime;
+            long nativeBuildTime = Utilities.getApkEntryBuildTime(RESOURCES_BUNDLE, this.cordova.getActivity());
+            if (nativeBuildTime != -1) {
+                String currentAppTimeStamp = String.valueOf(nativeBuildTime);
+                if ((deployedPackageTimeStamp != null) && (currentAppTimeStamp != null)) {
```

Comment:
also i noticed you added a comment in the iOS native code around here describing that this implies the package and native version are incompatible. i don't know if it's necessary but if you're going to keep it i'd put it here as well for consistency's sake